src/Shipment.php

```diff
@@ -0,0 +1,352 @@
+     */
+    public function createShipment(
+        array $sender,
+        array $destination,
+        array $parcel,
+        array $options = []
+    ) {
+        $content = $this->buildShipmentArray($sender, $destination, $parcel, $options);
+
+        $xml = Array2XML::createXML('non-contract-shipment', $content);
+        $envelope = $xml->documentElement;
+        $envelope->setAttribute('xmlns',
+            'http://www.canadapost.ca/ws/shipment-v8');
+        $payload = $xml->saveXML();
+
+        $response = $this->post(
+            "rs/{$this->customerNumber}/{$this->customerNumber}/shipment",
+            [
+                'Accept' => 'application/vnd.cpc.shipment-v8+xml',
+                'Content-Type' => 'application/vnd.cpc.shipment-v8+xml',
+            ],
+            $payload,
+            $options
+        );
+        return $response;
+    }
+
+    /**
+     * Get Shipment from Canada Post.
+     *
+     * @param string $shipment_id
+     *   The shipment id.
+     * @param string $extra
+     *   Gets the specific details for a shipment. Either 'details' or 'receipt'.
+     * @param array $options
+     *   The options array.
+     *
+     * @return \DOMDocument|\Psr\Http\Message\StreamInterface
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function getShipment($shipment_id, $extra = '', array $options = [])
+    {
+        $response = $this->get(
+            "rs/{$this->customerNumber}/{$this->customerNumber}/shipment/{$shipment_id}/{$extra}",
+            ['Accept' => 'application/vnd.cpc.shipment-v8+xml'],
+            $options
+        );
+        return $response;
+
+    }
+
+    /**
+     * Get Shipments from Canada Post within the specified range.
+     *
+     * @param string $from
+     *   The beginning range. YmdHs format, eg. 201808282359.
+     * @param string $to
+     *   The end range, defaults to current time. YmdHs format, eg. 201808282359.
+     * @param string $tracking_pin
+     *   The Tracking PIN of the shipment to retrieve.
+     * @param array $options
+     *   The options array.
+     *
+     * @see https://www.canadapost.ca/cpo/mc/business/productsservices/developers/services/shippingmanifest/shipments.jsf
+     * @return \DOMDocument|\Psr\Http\Message\StreamInterface
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function getShipments($from, $to = '', $tracking_pin = '', array $options = [])
+    {
+        // TODO convert to shipment API
+        if (empty($to)) {
+            $to = date('YmdHs');
+        }
+        $query_params = "from={$from}&to{$to}";
+
+        if (!empty($tracking_pin)) {
+            $query_params = "trackingPIN={$tracking_pin}";
+        }
+
+        $response = $this->get(
+            "rs/{$this->config['customer_number']}/{$this->config['customer_number']}/shipment?" . $query_params,
+            ['Accept' => 'application/vnd.cpc.shipment-v8+xml'],
+            $options
+        );
```

Comment:
You can avoid long lines in such cases using `sprintf`:

```
$endpoint = sprintf(
  'rs/%s/%s/shipment?%s',
  {$this->config['customer_number']},
  {$this->config['customer_number']},
  $query_params
);
```

Even though not a big deal in this particular case, just the usual comment that as far as readability is concerned it's better to expand vertically than horizontally.

Also, am I seeing correctly, the URL requires the customer number to be passed twice?